subsys/bluetooth/host/att.c

```diff
@@ -507,6 +525,8 @@ static void chan_rsp_sent(struct bt_conn *conn, void *user_data, int err)
 		atomic_clear_bit(chan->flags, ATT_PENDING_RSP);
 	}
 
+	chan->rsp_buf = NULL;
```

Comment:
Missing `net_buf_unref(chan->rsp_buf)` before this. Please don't introduce this kind of implicit reference counting where you play around with reference counted pointers without using `ref()` and `unref(). It makes the code difficult to follow and review, and risks introducing bugs.